### PR TITLE
added logic to set port speed in primary nic

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISFloatingIPDataSource_basic (468.64s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 472.872s
```

### CODE REVIEW

**Comment by astha-jain** [comment]
Still not seeing the port speed in response.  Compare CLI and terraform state file 
CLI
```
 "primary_network_interface": {
        "allow_ip_spoofing": false,
        "created_at": "2021-11-11T17:53:51.000Z",
        "floating_ips": null,
        "href": "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instances/0726_ed70e8f8-b314-4714-9082-169ebcec036c/network_interfaces/0726-88693063-e69e-4499-93b8-70c93c4c9121",
        "id": "0726-88693063-e69e-4499-93b8-70c93c4c9121",
        "ips": null,
        "name": "debunk-sift-cabbie-frisk",
        "port_speed": 1500,
        "primary_ipv4_address": "10.240.68.4",
        "resource_type": "network_interface",
        "security_groups": [
            {
                "crn": "crn:v1:staging:public:is:us-south:a/3bf43234b648447d95aa1aee1bc5e73a::security-group:r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a",
                "href": "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/security_groups/r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a",
                "id": "r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a",
                "name": "clenched-bridged-waggle-waving"
            }
        ],
        "status": "available",
        "subnet": {
            "crn": "crn:v1:staging:public:is:us-south-2:a/3bf43234b648447d95aa1aee1bc5e73a::subnet:0726-9c1f3d01-fd3c-4090-87d0-5e7b997a5799",
            "href": "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/subnets/0726-9c1f3d01-fd3c-4090-87d0-5e7b997a5799",
            "id": "0726-9c1f3d01-fd3c-4090-87d0-5e7b997a5799",
            "name": "tfins-subnet-90"
        },
        "type": "primary"
    },
```

 and  terraform state

```
            "primary_network_interface": [
              {
                "id": "0726-88693063-e69e-4499-93b8-70c93c4c9121",
                "name": "debunk-sift-cabbie-frisk",
                "port_speed": 0,
                "primary_ipv4_address": "10.240.68.4",
                "security_groups": [
                  "r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a"
                ],
                "subnet": "0726-9c1f3d01-fd3c-4090-87d0-5e7b997a5799"
              }
            ],
```

**Comment by astha-jain** [comment]
Also doc update is required to remove deprecated attribute

File: website/docs/r/is_instance.html.markdown [hunk 1]
```diff
@@ -295,7 +295,6 @@ Review the argument references that you can specify for your resource.
       - Use this only if you have **IP spoofing operator** access.
 
   - `name` - (Optional, String) The name of the network interface.
-  - `port_speed` - (Deprecated, Integer) Speed of the network interface.
```
Comment (astha-jain) [inline website/docs/r/is_instance.html.markdown:298]: With this removal, lets make sure we remove the argument support from datasource https://github.com/ibm-vpc/terraform-provider-ibm/blob/7d21dbe45b7beb4b962702a4c63eee5bde9c1df0/ibm/resource_ibm_is_instance.go#L317

**Comment by ujjwal-ibm** [comment]
rebased and removed port speed from test and document

**Comment by ujjwal-ibm** [comment]
closing, moved to upstream

**Pull request closed without merging**